include/modem/location.h

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file location.h
+ * @brief Public APIs for the Location library.
+ * @defgroup location Location library
+ * @{
+ */
+
+#ifndef LOCATION_H_
+#define LOCATION_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define LOC_MAX_METHODS 3
+
+/** Positioning methods. */
+enum loc_method {
+	/** LTE cellular positioning. */
+	LOC_METHOD_CELLULAR = 1,
+	/** Global Navigation Satellite System (GNSS). */
+	LOC_METHOD_GNSS,
+	/** WLAN positioning. */
+	LOC_METHOD_WLAN,
+};
+
+/** Event IDs. */
+enum loc_event_id {
+	/** Location update. */
+	LOC_EVT_LOCATION,
+	/** Getting location timed out. */
+	LOC_EVT_TIMEOUT,
+	/** An error occurred when trying to get the location. */
+	LOC_EVT_ERROR
+};
+
+/** Positioning accuracy enumerator. */
+enum loc_accuracy {
+	/** Allow lower accuracy to conserve power. */
+	LOC_ACCURACY_LOW,
+	/** Normal accuracy. */
+	LOC_ACCURACY_NORMAL,
+	/** Allow higher power consumption to improve accuracy. */
+	LOC_ACCURACY_HIGH
+};
+
+/** WLAN positioning service. */
+enum loc_wlan_service {
```

Comment:
As discussed, we should think of a default service if only one is configured on so that client request don't need to have anything. Not sure if this makes sense but I'll add a note here so we'll discuss it in the future.